controller/tenant.go

```diff
@@ -175,6 +176,9 @@ func (c *TenantController) Setup(ctx *app.SetupTenantContext) error {
 		}
 		err = tenantRepository.CreateTenant(dbTenant)
 		if err != nil {
+			if strings.Contains(err.Error(), "pq: duplicate key value violates unique constraint") {
```

Comment:
Don't we have better way to check if there is such error rather checking error_msg_string.
OR
Can we first call something like below ??
```
if repo.GetTenant(tenantId) != nil {
  return ctx.Conflic()
} else {
  repo.CreateTenant()
}
```

Reply:
> Don't we have better way to check if there is such error rather checking error_msg_string.

I would love to have, but gorm doesn't provide such an error kind: https://github.com/jinzhu/gorm/blob/master/errors.go

> Can we first call something like below ??

I check for the existence already https://github.com/fabric8-services/fabric8-tenant/pull/755/files/32679607ef9ce58f269236b6dcc6b36ec48d7d96#diff-dfb0ecdc519ac765cb68432d078cc1d0R143 but this doesn't work for two POST calls that run at the same time concurrently - in that case, I would need to lock the table